### PR TITLE
Refactor external table error handling

### PR DESCRIPTION
**Pull request by hlinnaka**
To use Resource Owners to make sure that curl and file handles are closed on abort. See also discussion on gpdb-dev, https://groups.google.com/a/greenplum.org/forum/#!topic/gpdb-dev/GnrYQR0-KA8

### CODE REVIEW

File: src/backend/access/external/url_curl.c [hunk 1]
```diff
@@ -1662,95 +1631,67 @@ gp_proto1_read(char *buf, int bufsz, URL_FILE *file, CopyState pstate, char *buf
...
+	set_httpheader(file, "X-GP-DONE", "1");
 
 	/* use empty message */
-	if (CURLE_OK != (e = curl_easy_setopt(file->u.curl.handle, CURLOPT_POSTFIELDS, "")))
-	{
-		elog(WARNING, "internal error: curl_easy_setopt CURLOPT_POSTFIELDS %s error (%d - %s)",
-			 file->u.curl.curl_url,
+	if (CURLE_OK != (e = curl_easy_setopt(file->curl->handle, CURLOPT_POSTFIELDS, "")))
+		elog(ERROR, "internal error: curl_easy_setopt CURLOPT_POSTFIELDS %s error (%d - %s)",
+			 file->curl_url,
```
Comment (yydzero) [inline src/backend/access/external/url_curl.c]: Use CURL_EASY_SETOPT()?

Comment (hlinnaka) [inline src/backend/access/external/url_curl.c]: Ah yeah, makes sense, thanks. Changed to use CURL_EASY_SETOPT

**Comment by yydzero** [comment]
LGTM, Great improvement!

**Comment by yuanzhao** [comment]
👍 

LGTM, also did some tests, positive.

**Comment by hlinnaka** [comment]
Ok, pushed. Thanks for the review!

**Pull request closed without merging**